### PR TITLE
Changed Dispatcher.map to a default dict to support multiple mappings

### PR DESCRIPTION
**Pull request by jsheedy**
I wanted to be able to map addresses to more than one handler, and found this TODO in the dispatcher:

>  Check if we need to use a multimap instead, spec is a bit fuzzy about it...

It's not clear to me either whether a dispatcher is required by the spec to allow multiple mappings of the same address. But it made sense in my use case, so this pull request enables it.  So now handler1 and handler2 are both called by this dispatcher:

```Python
dispatcher.map("/foo/bar", handler1)
dispatcher.map("/foo/bar", handler2)
```
I used a defaultdict(list) instead of a multimap, since its in the standard library.

I changed handlers_for_address to return a generator rather than a list of handlers since the list comprehension was getting a little long to read quickly. This meant a few more of the dispatcher tests needed to use the sortAndAssertSequenceEqual (or something, to convert from a generator to a list for assertion), but the tests all pass fine.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+__pycache__
```
Comment (attwad) [inline .gitignore]: If we add a gitignore I'd go with this one personally
https://github.com/github/gitignore/blob/master/Python.gitignore

File: pythonosc/dispatcher.py [hunk 2]
```diff
@@ -44,17 +42,17 @@ def handlers_for_address(self, address_pattern):
...
-        handler for addr, handler in self._map.items()
-        if ( pattern.match(addr) # wildcards could be in the incoming pattern
-             # .. or the mapped pattern
-             or (('*' in addr) and re.match(addr.replace('*','[^/]*?/*'), address_pattern))
-           )
-        ]
+    matched = False
+
+    for addr, handlers in self._map.items():
+      if pattern.match(addr) \
```
Comment (attwad) [inline pythonosc/dispatcher.py]: just a style concern but I prefer avoiding the use of \, prefer (instead.

if (pattern.match(addr)
    or (('*' in addr) and re.match(addr.replace('*','[^/]*?/*'), address_pattern)):

File: pythonosc/dispatcher.py [hunk 3]
```diff
@@ -25,9 +25,7 @@ def map(self, address, handler, *args):
       - args: Any additional arguments that will be always passed to the
               handlers after the osc messages arguments if any.
     """
-    # TODO: Check if we need to use a multimap instead, spec is a bit fuzzy
-    # about it...
-    self._map[address] = Handler(handler, list(args))
+    self._map[address].append(Handler(handler, list(args)))
 
   def handlers_for_address(self, address_pattern):
     """Return a tuple of Handler namedtuple matching the given OSC pattern."""
```
Comment (attwad) [inline pythonosc/dispatcher.py]: Can you update the comment please?

**Comment by jsheedy** [comment]
Thanks for the feedback, the latest commit makes the changes
- switched to suggested .gitignore
- updated spec comment
- removed \

File: pythonosc/dispatcher.py [hunk 4]
```diff
@@ -25,9 +25,10 @@ def map(self, address, handler, *args):
...
-    # TODO: Check if we need to use a multimap instead, spec is a bit fuzzy
-    # about it...
-    self._map[address] = Handler(handler, list(args))
+    # TODO: Check the spec:
+    # http://opensoundcontrol.org/spec-1_0
+    # regarding multiple mappings
+    self._map[address].append(Handler(handler, list(args)))
 
   def handlers_for_address(self, address_pattern):
     """Return a tuple of Handler namedtuple matching the given OSC pattern."""
```
Comment (attwad) [inline pythonosc/dispatcher.py]: did you forget to include the comment change? this function now yields, please update the comment accordingly.

Comment (jsheedy) [inline pythonosc/dispatcher.py]: My bad, I updated the comment and not the docstring.  Updated now.

**Comment by attwad** [comment]
alright good to go! thanks a lot!

**Pull request merged**